### PR TITLE
Rule S1134: Track uses of "FIXME" tags

### LINKED ISSUE #92: Rule S1134: Track uses of "FIXME" tags
Implements [RSPEC-1134](https://jira.sonarsource.com/browse/RSPEC-1134)

### PR DESCRIPTION
**Pull request by stylianos-agapiou-sonarsource**
Fixes #92

### CODE REVIEW

File: common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/FixMeTagPresenceCheck.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    super(UastNode.Kind.COMMENT, UastNode.Kind.STRUCTURED_COMMENT);
+  }
+
+  @Override
+  public void visitNode(UastNode node) {
+    commentContainsPattern(node, "FIXME");
+  }
+
+  private void commentContainsPattern(UastNode node, String pattern) {
+    String comment = node.token.value.toUpperCase(Locale.ENGLISH);
```
Comment (stylianos-agapiou-sonarsource) [inline common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/FixMeTagPresenceCheck.java]: `token` is Nullable!

File: common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/FixMeTagPresenceCheck.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    if (comment.contains(pattern) && !isLetterAround(comment, pattern)) {
+      reportIssue(node, "Take the required action to fix the issue indicated by this \"FIXME\" comment.");
+    }
+  }
+
+  private static boolean isLetterAround(String comment, String pattern) {
```
Comment (stylianos-agapiou-sonarsource) [inline common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/FixMeTagPresenceCheck.java]: Use 'java Pattern' instead

File: common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/FixMeTagPresenceCheck.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  public FixMeTagPresenceCheck() {
+    super(UastNode.Kind.COMMENT, UastNode.Kind.STRUCTURED_COMMENT);
+  }
+
+  @Override
+  public void visitNode(UastNode node) {
+    commentContainsPattern(node, "FIXME");
+  }
+
+  private void commentContainsPattern(UastNode node, String pattern) {
```
Comment (stylianos-agapiou-sonarsource) [inline common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/FixMeTagPresenceCheck.java]: inline method

File: common-rule-engine/src/test/files/checks/FixMeTagPresenceCheck/FixMeTagPresenceCheck.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package main
+
```
Comment (stylianos-agapiou-sonarsource) [inline common-rule-engine/src/test/files/checks/FixMeTagPresenceCheck/FixMeTagPresenceCheck.go]: Add message.

File: common-rule-engine/src/test/files/checks/FixMeTagPresenceCheck/FixMeTagPresenceCheck.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+// Noncompliant@+1
+// fixME
+class A {
+  // Noncompliant@+1
+  /*
+   * fixme
+   * FixMe
+   */
+  //
+  // Noncompliant@+1 [[sc=3;ec=11]]
```
Comment (stylianos-agapiou-sonarsource) [inline common-rule-engine/src/test/files/checks/FixMeTagPresenceCheck/FixMeTagPresenceCheck.java]: wrong assertion syntax

File: common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/FixMeTagPresenceCheck.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import org.sonar.uast.UastNode;
+
+@Rule(key = "S1134")
+public class FixMeTagPresenceCheck extends Check {
+
+  public FixMeTagPresenceCheck() {
+    super(UastNode.Kind.COMMENT, UastNode.Kind.STRUCTURED_COMMENT);
+  }
+
+  private static final Pattern fixMePattern = Pattern.compile("\\WFIXME(\\W|$)");
```
Comment (vilchik-elena) [inline common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/FixMeTagPresenceCheck.java]: @stylianos-agapiou-sonarsource move and rename

**Review by vilchik-elena — APPROVED** [review]

**Pull request merged**